models/repo.go

```diff
@@ -264,6 +264,11 @@ func (repo *Repository) HTMLURL() string {
 	return setting.AppURL + repo.FullName()
 }
 
+// APIURL returns the repository API URL
+func (repo *Repository) APIURL() string {
+	return strings.TrimRight(setting.AppURL, "/") + "/" + path.Join("api/v1/repos", repo.FullName())
```

Comment:
I don't see how that is relevant, it's supposed to be an absolute URL, not a relative one.

Works just fine IMO: https://play.golang.org/p/Rc734RIuJu